validator/accounts/v2/accounts_backup.go

```diff
@@ -0,0 +1,267 @@
+	}
+
+	// Input the directory where they wish to backup their accounts.
+	backupDir, err := inputDirectory(cliCtx, backupPromptText, flags.BackupDirFlag)
+	if err != nil {
+		return errors.Wrap(err, "could not parse keys directory")
+	}
+
+	// Allow the user to interactively select the accounts to backup or optionally
+	// provide them via cli flags.
+	filteredPubKeys, err := determinePublicKeysForBackup(cliCtx, pubKeys)
+	if err != nil {
+		return errors.Wrap(err, "could not filter public keys for backup")
+	}
+
+	// Ask the user for their desired password for their backed up accounts.
+	backupsPassword, err := promptutil.InputPassword(
+		cliCtx,
+		flags.BackupsPasswordFile,
+		"Enter a new password for your backed up accounts",
+		"Confirm new password",
+		promptutil.ConfirmPass,
+		promptutil.ValidatePasswordInput,
+	)
+	if err != nil {
+		return errors.Wrap(err, "could not determine password for backed up accounts")
+	}
+
+	var keystoresToBackup []*v2keymanager.Keystore
+	switch wallet.KeymanagerKind() {
+	case v2keymanager.Direct:
+		km, ok := keymanager.(*direct.Keymanager)
+		if !ok {
+			return errors.New("could not assert keymanager interface to concrete type")
+		}
+		keystoresToBackup, err = km.ExtractKeystores(ctx, filteredPubKeys, backupsPassword)
+		if err != nil {
+			return errors.Wrap(err, "could not backup accounts for direct keymanager")
+		}
+	case v2keymanager.Derived:
+		km, ok := keymanager.(*derived.Keymanager)
+		if !ok {
+			return errors.New("could not assert keymanager interface to concrete type")
+		}
+		_ = km
+		return nil
+	default:
+		return errors.New("keymanager kind not supported")
+	}
+	return zipKeystoresToOutputDir(keystoresToBackup, backupDir)
+}
+
+func determinePublicKeysForBackup(cliCtx *cli.Context, validatingPublicKeys [][48]byte) ([]bls.PublicKey, error) {
+	var filteredPubKeys []bls.PublicKey
+	if cliCtx.IsSet(flags.BackupForPublicKeysFlag.Name) {
+		pubKeyStrings := strings.Split(cliCtx.String(flags.BackupForPublicKeysFlag.Name), ",")
+		if len(pubKeyStrings) == 0 {
+			return nil, fmt.Errorf(
+				"could not parse %s. It must be a string of comma-separated hex strings",
+				flags.BackupForPublicKeysFlag.Name,
+			)
+		}
+		for _, str := range pubKeyStrings {
+			pkString := str
+			if strings.Contains(pkString, "0x") {
+				pkString = pkString[2:]
+			}
+			pubKeyBytes, err := hex.DecodeString(pkString)
+			if err != nil {
+				return nil, errors.Wrapf(err, "could not decode string %s as hex", pkString)
+			}
+			blsPublicKey, err := bls.PublicKeyFromBytes(pubKeyBytes)
+			if err != nil {
+				return nil, errors.Wrapf(err, "%#x is not a valid BLS public key", pubKeyBytes)
+			}
+			filteredPubKeys = append(filteredPubKeys, blsPublicKey)
+		}
+		return filteredPubKeys, nil
+	}
+	return selectAccounts(cliCtx, validatingPublicKeys)
+}
+
+// Ask user to select accounts via an interactive prompt.
+func selectAccounts(cliCtx *cli.Context, pubKeys [][48]byte) ([]bls.PublicKey, error) {
```

Comment:
This function is also used with `accounts-v2 delete`. can you make sure everything is as expected for it?